tests/hardhat/Fork/XVSVaultUpgradeTimeManager.ts

```diff
@@ -0,0 +1,64 @@
+import { smock } from "@defi-wonderland/smock";
+import { impersonateAccount } from "@nomicfoundation/hardhat-network-helpers";
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
+import chai from "chai";
+import { ethers, network } from "hardhat";
+
+import { XVSVault, XVSVaultProxy } from "../../../typechain";
+
+const { expect } = chai;
+chai.use(smock.matchers);
+
+const XVS_VAULT_PROXY = "0x9aB56bAD2D7631B2A857ccf36d998232A8b82280";
+const TIMELOCK = "0xce10739590001705F7FF231611ba4A48B2820327";
+
+export async function setForkBlock(blockNumber: number) {
+  await network.provider.request({
+    method: "hardhat_reset",
+    params: [
+      {
+        forking: {
+          jsonRpcUrl: process.env[`ARCHIVE_NODE_${process.env.FORKED_NETWORK}`],
+          blockNumber: blockNumber,
+        },
+      },
+    ],
+  });
+}
+
+const forking = (blockNumber: number, fn: () => void) => {
+  describe(`Reward and Setter facet upgrade check At block #${blockNumber}`, () => {
+    before(async () => {
+      await setForkBlock(blockNumber);
+    });
+    fn();
+  });
+};
```

Comment:
We can import these functions from [utils.ts](https://github.com/VenusProtocol/venus-protocol/blob/7d3e10178e51982a7474a2831fc38e8f89a0faeb/tests/hardhat/Fork/utils.ts)